pylint_odoo/checkers/modules_odoo.py

```diff
@@ -568,13 +568,24 @@ def _get_duplicate_xml_fields(self, fields):
         """
         all_fields = {}
         for field in fields:
-            field_xml = field.attrib.get('name')
-            if not field_xml:
+            name = field.attrib.get('name')
+            context = field.attrib.get('context')
+            filter_domain = field.attrib.get('filter_domain')
+            sourceline = field.sourceline
+            if not name:
                 continue
-            all_fields.setdefault(
-                (field_xml, field.getparent()), []).append(field)
+            for in_field in fields:
```

Comment:
The following logic
```python
for field in fields:
    ...
   for in_field in fields:
```

It's so slow and so bad for performance.
Because you are running loops `n^n`

What about use a complex key for all_fields dict?

I mean,
```diff
- all_fields.setdefault((name, field.getparent()), []).append(field)
+ all_fields.setdefault((name, context, filter_domain, field.getparent()), []).append(field) 
```

If all keys values are duplicated so the items will be more than 2. (Like as original logic)

Reply:
@moylop260 
You're right again, Thanks for help me in this topic